src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.AllLowerCamelCaseMatcher.cs

```diff
@@ -118,27 +127,40 @@ public AllLowerCamelCaseMatcher(string candidate, bool includeMatchedSpans, Stri
                             // We found a path that allowed us to match everything contiguously
                             // from the beginning.  This is the best match possible.  So we can
                             // just stop now and return this result.
-                            return (weight, matchedSpans);
+                            //
+                            // Return any previous best results back to the pool.
+                            bestMatchedSpansInReverse?.Free();
```

Comment:
:bulb: If you move the containing block below the next block, you can remove the special handling this call to `Free`, and also just keep the original comment. If you do this, the return statement would be changed to:

```csharp
return (weight, bestMatchedSpansInReverse);
```